controllers/core/node_controller.go

```diff
@@ -45,12 +45,13 @@ type NodeReconciler struct {
 func (r *NodeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
```

Comment:
A comment about the situations in which UpdateNode, and AddNode is done, in the reconcile method might be helpful here. It seems like an important behavior change in reconcile.